iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/PublicNetworkAccessCheck.java

```diff
@@ -32,14 +35,128 @@ public class PublicNetworkAccessCheck extends AbstractArmResourceCheck {
+    register("Microsoft.SecurityAndCompliance/privateLinkServicesForSCCPowershell", checkPublicNetworkAccessSimplified());
+    register("Microsoft.ServiceBus/namespaces", checkPublicNetworkAccessSimplified());
+    register("Microsoft.ServiceBus/namespaces/networkRuleSets", checkPublicNetworkAccessSimplified());
+    register("Microsoft.ServiceBus/namespaces/networkRuleSets", checkPublicNetworkAccessSimplified());
+    register("Microsoft.SignalRService/signalR", checkPublicNetworkAccessSimplified());
+    register("Microsoft.SignalRService/signalR", checkPublicNetworkAccessSimplified());
+    register("Microsoft.SignalRService/webPubSub", checkPublicNetworkAccessSimplified());
+    register("Microsoft.SignalRService/webPubSub", checkPublicNetworkAccessSimplified());
+    register("Microsoft.Sql/servers", checkPublicNetworkAccessSimplified());
+    register("Microsoft.Sql/servers", checkPublicNetworkAccessSimplified());
+    register("Microsoft.Storage/storageAccounts", checkPublicNetworkAccessSimplified());
+    register("Microsoft.Synapse/workspaces", checkPublicNetworkAccessSimplified());
+    register("Microsoft.Synapse/workspaces", checkPublicNetworkAccessSimplified());
+    register("Microsoft.TimeSeriesInsights/environments", checkPublicNetworkAccessSimplified());
+    register("Microsoft.Web/sites", checkPublicNetworkAccessSimplified());
+    register("Microsoft.Web/sites/config", checkPublicNetworkAccessSimplified());
+    register("Microsoft.Web/sites/slots", checkPublicNetworkAccessSimplified());
+    register("Microsoft.Web/sites/slots/config", checkPublicNetworkAccessSimplified());
+    register("Microsoft.Web/staticSites", checkPublicNetworkAccessSimplified());
+  }
+
+  private static BiConsumer<CheckContext, ResourceDeclaration> checkPublicNetworkAccess() {
+    return (ctx, resource) -> PropertyUtils.value(resource, "publicNetworkAccess")
+      .filter(PublicNetworkAccessCheck::isSensitivePublicNetworkAccess)
+      .ifPresent(value -> ctx.reportIssue(value, "Make sure allowing public network access is safe here."));
   }
 
-  private static boolean isSensitiveStringLiteral(Tree tree) {
+  private static BiConsumer<CheckContext, ResourceDeclaration> checkPublicNetworkAccessSimplified() {
+    return (ctx, resource) -> PropertyUtils.value(resource, "publicNetworkAccess")
+      .filter(PublicNetworkAccessCheck::isSensitivePublicNetworkAccessSimplified)
+      .ifPresent(value -> ctx.reportIssue(value, "Make sure allowing public network access is safe here."));
+  }
+
+  private static boolean isSensitivePublicNetworkAccess(Tree tree) {
     return ((ArmTree) tree).is(ArmTree.Kind.STRING_LITERAL) && SENSITIVE_VALUES.contains(((StringLiteral) tree).value());
   }
+
+  private static boolean isSensitivePublicNetworkAccessSimplified(Tree tree) {
+    return ((ArmTree) tree).is(ArmTree.Kind.STRING_LITERAL) && "Enabled".equals(((StringLiteral) tree).value());
+  }
```

Comment:
```suggestion
  private static boolean isSensitivePublicNetworkAccess(Tree tree) {
    return TextUtils.matchesValue(tree, SENSITIVE_VALUES::contains).isTrue();
  }

  private static boolean isSensitivePublicNetworkAccessSimplified(Tree tree) {
    return TextUtils.matchesValue(tree, "Enabled"::contains).isTrue();
  }
```
Suggesting the use of `TextUtils`: our `StringLiteral` complies to its contract and it simplify the checks.